File: mobile/hika_app/lib/features/trust_safety/data/report.dart
/// Mirrors backend Hika.Domain.TrustSafety.ReportReason — serialized by member name.
enum ReportReason {
  harassment('Harassment', 'Harassment'),
  unsafeDriving('UnsafeDriving', 'Unsafe driving'),
  noShow('NoShow', 'No-show'),
  scam('Scam', 'Scam or fraud'),
  other('Other', 'Something else');

  const ReportReason(this.wireValue, this.displayName);

  final String wireValue;
  final String displayName;
}

/// Mirrors backend Hika.Application.TrustSafety.Dtos.ReportResponse.
class Report {
  const Report({
    required this.id,
    this.reportedUserId,
    this.reportedTripId,
    required this.reason,
    required this.description,
    required this.status,
    required this.createdAtUtc,
  });

  factory Report.fromJson(Map<String, dynamic> json) => Report(
    id: json['id'] as String,
    reportedUserId: json['reportedUserId'] as String?,
    reportedTripId: json['reportedTripId'] as String?,
    reason: json['reason'] as String,
    description: json['description'] as String,
    status: json['status'] as String,
    createdAtUtc: DateTime.parse(json['createdAtUtc'] as String),
  );

  final String id;
  final String? reportedUserId;
  final String? reportedTripId;
  final String reason;
  final String description;
  final String status;
  final DateTime createdAtUtc;
}

File: backend/src/Hika.Domain/TrustSafety/Report.cs
using Hika.Domain.Common;

namespace Hika.Domain.TrustSafety;

public enum ReportReason
{
    Harassment,
    UnsafeDriving,
    NoShow,
    Scam,
    Other,
}

public enum ReportStatus
{
    Open,
    UnderReview,
    Resolved,
    Dismissed,
}

/// <summary>A user or trip flagged for review. At least one of ReportedUserId/ReportedTripId is
/// always set — enforced by CreateReportRequestValidator, not a DB constraint, since "at least
/// one of two nullable columns" isn't cleanly expressible relationally.</summary>
public sealed class Report : AuditableEntity
{
    public Guid ReporterUserId { get; private set; }

    public Guid? ReportedUserId { get; private set; }

    public Guid? ReportedTripId { get; private set; }

    public ReportReason Reason { get; private set; }

    public string Description { get; private set; }

    public ReportStatus Status { get; private set; }

    private Report()
    {
        Description = string.Empty;
    }

    public static Report File(Guid reporterUserId, Guid? reportedUserId, Guid? reportedTripId, ReportReason reason, string description) =>
        new()
        {
            ReporterUserId = reporterUserId,
            ReportedUserId = reportedUserId,
            ReportedTripId = reportedTripId,
            Reason = reason,
            Description = description,
            Status = ReportStatus.Open,
        };
}
